composer/callbacks/checkpoint_saver.py

```diff
@@ -289,25 +289,26 @@ class CheckpointSaver(Callback):  # noqa: D101
 
     def __init__(
         self,
-        folder: str = '{run_name}/checkpoints',
-        filename: str = 'ep{epoch}-ba{batch}-rank{rank}.pt',
+        checkpoint_save_path: str = '{run_name}/checkpoints',
+        checkpoint_filename: str = 'ep{epoch}-ba{batch}-rank{rank}.pt',
         artifact_name: Optional[str] = '{run_name}/checkpoints/ep{epoch}-ba{batch}-rank{rank}',
-        latest_filename: Optional[str] = 'latest-rank{rank}.pt',
+        latest_checkpoint_filename: Optional[str] = 'latest-rank{rank}.pt',
         latest_artifact_name: Optional[str] = '{run_name}/checkpoints/latest-rank{rank}',
-        save_interval: Union[Time, str, int, Callable[[State, Event], bool]] = '1ep',
+        checkpoint_save_interval: Union[Time, str, int, Callable[[State, Event], bool]] = '1ep',
         *,
         overwrite: bool = False,
         num_checkpoints_to_keep: int = -1,
         weights_only: bool = False,
     ):
-        if not callable(save_interval):
-            save_interval = checkpoint_periodically(save_interval)
-        self.save_interval = save_interval
+        if not callable(checkpoint_save_interval):
+            checkpoint_save_interval = checkpoint_periodically(checkpoint_save_interval)
+        self.checkpoint_save_interval = checkpoint_save_interval
 
-        self.folder = folder
+        self.checkpoint_save_path = checkpoint_save_path
 
-        self.filename = PartialFilePath(filename.lstrip('/'), folder)
-        self.latest_filename = PartialFilePath(latest_filename.lstrip('/'), folder) if latest_filename else None
+        self.checkpoint_filename = PartialFilePath(checkpoint_filename.lstrip('/'), checkpoint_save_path)
+        self.latest_checkpoint_filename = PartialFilePath(latest_checkpoint_filename.lstrip('/'),
```

Comment:
minor nit: can we rename this to `checkpoint_path` and `latest_checkpoint_path`? Otherwise, the `self.checkpoint_filename.filename` is a bit weird

Reply:
yeah this does get confusing. it is even more confusing in part 2 of the PR series. will consider it for the last PR in the series